homeassistant/components/openai_conversation/__init__.py

```diff
@@ -27,32 +38,77 @@
     DEFAULT_PROMPT,
     DEFAULT_TEMPERATURE,
     DEFAULT_TOP_P,
+    DOMAIN,
 )
 
 _LOGGER = logging.getLogger(__name__)
+SERVICE_GENERATE_IMAGE = "generate_image"
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up OpenAI Conversation."""
+
+    async def render_image(call: ServiceCall) -> ServiceResponse:
+        """Render an image with dall-e."""
+        try:
+            response = await openai.Image.acreate(
```

Comment:
I can't look into this call but does it time-out on its own or should that be guarded ?

Reply:
It has a default timeout defined in the library of 600 seconds